### ISSUE TITLE
Fail if a Test contains another Test

### ISSUE THREAD

**Issue by robstoll** [JetBrains/spek#371]
I accidentally wrapped a test into another test and only realised it because I used TDD and the Test did not fail as expected. I am not using TDD all the time and would like to be informed of such a mistake by Spek. So, the test `it has a test in it` should fail at the end because it contained another test.
```
object Spec: Spek({
    it("has a test in it") {
        test("foo") {
            throw IllegalStateException("will never be shown")
        }
    }
})
```

**Comment by raniejade** [comment]
Related to #331, using `@DslMarker` should fix this as well.

**Comment by raniejade** [comment]
Resolved by #398
